source/common/ur_pool_manager.hpp

```diff
@@ -158,63 +194,83 @@ inline bool pool_descriptor::operator==(const pool_descriptor &other) const {
         }
     }
 
+    if (lhs.allocLocation.has_value()) {
+        if (!rhs.allocLocation.has_value()) {
+            return false;
+        }
+
+        if (lhs.allocLocation.value().location !=
+            rhs.allocLocation.value().location) {
+            return false;
+        }
+    }
+
     return lhsNative == rhsNative && lhs.type == rhs.type &&
```

Comment:
What about isSharedAllocationReadOnlyOnDevice? Is there no distinction between shared and sharedReadOnly pools anymore?

Reply:
the distinction is now encoded in `deviceFlags`, the only difference before was that ReadOnlyOnDevice pools passed the `UR_USM_DEVICE_MEM_FLAG_DEVICE_READ_ONLY` flag to the shared allocation entry point

Reply:
Right, that makes sense.